docs/reference/badge_spec.md

```diff
@@ -0,0 +1,95 @@
+
+Below is the specification for a badge, the data it may contain, the valid values that may be placed in an enum, and what should be done with it.
+
+## Schema
+
+The following schema is required of any `generic` badge type:
+
+```json
+"badges": [
+  {
+    "type": "<type_enum>",
+    "title": "<string>",
+    "text": "<string>",
+    "link": "<string>"
+  }
+]
+```
+
+As you can see `badges` is an array of a valid `badge` object.
+A side note, while there is currently no limit for the amount of badges a single package may contain, the current recommendation is to try and stay around 5 badges or less. This may be later on solidified into a proper part of the schema if required.
+
+Any `badge` object may contain the following properties:
+
+* `title`: This is a **required** property. Specifying the title of the badge. Such as `Deprecated`. This value is a strict enum, with only the values specified below being allowed.
+* `text`: This is an **optional** property. Specifying further information about this badge. That may be hidden either based on a user setting or by context.
+* `link`: This is an **optional** property. Providing a link for the user to be directed to, that may contain further information about the badge.
+* `type`: This is a **required** property. Instructing what type of badge this is. The type of badge should be used to determine any icons used alongside the badge, as well as any color or styling the badge will receive when displayed by the client. The valid `type`s currently available are as follows:
+  - `warn`: This should be used to indicate something that the user **must** be aware of.
+  - `info`: This should be used to point out information that is neutral in the user receiving.
+  - `success`: Should be used to indicate something good about the badge, that the user would be delighted to know about.
+  - A note about `type`: It's highly recommended for your client to support default styling of badges, in the event new types are added before your client has support for this type. Default styling will ensure badges will always appear.
+
+The following are some valid examples of badges:
+
+```json
+"badges": [
+  {
+    "type": "warn",
+    "title": "Deprecated",
+    "text": "This package is out of date and should not be used.",
+    "link": "https://github.com/pulsar-edit/pulsar/issues/1"
+  },
+  {
+    "type": "info",
+    "title": "Looking for Maintainers"
+  },
+  {
+    "type": "success",
+    "title": "Made for Pulsar!",
+    "link": "https://github.com/pulsar-edit/pulsar"
+  }
+]
+```
+
+# Types of Badges & Why They May be Added
```

Comment:
Ideally we shouldn't have more than one `h1` in a document. There should only be a single `h1` and the rest cascade from that.